### ISSUE TITLE
Intermittent NullPointerException when creating or deleting resources

### ISSUE THREAD

**Issue by sunmit9** [crnk-project/crnk-framework#483]
Encountering NPE intermittently when trying to create a new resource with a POST or DELETE api. Following is the log trace from crnk.

`{"exception":{"stacktrace":"java.lang.NullPointerException\n\tat io.crnk.core.repository.foward.strategy.SetOwnerStrategy.addRelations(SetOwnerStrategy.java:64)\n\tat io.crnk.core.repository.foward.ForwardingRelationshipRepository.addRelations(ForwardingRelationshipRepository.java:141)\n\tat io.crnk.core.engine.internal.repository.RelationshipRepositoryAdapterImpl$3.invoke(RelationshipRepositoryAdapterImpl.java:120)\n\tat io.crnk.core.engine.internal.repository.ResponseRepositoryAdapter$RepositoryRequestFilterChainImpl.doFilter(ResponseRepositoryAdapter.java:241)\n\tat io.crnk.core.engine.internal.repository.RelationshipRepositoryAdapterImpl.addRelations(RelationshipRepositoryAdapterImpl.java:129)\n\tat io.crnk.core.engine.internal.dispatcher.controller.FieldResourcePost.attachToParent(FieldResourcePost.java:122)\n\tat io.crnk.core.engine.internal.dispatcher.controller.FieldResourcePost.lambda$handleAsync$2(FieldResourcePost.java:88)\n\tat io.crnk.core.engine.result.ImmediateResult.zipWith(ImmediateResult.java:45)\n\tat io.crnk.core.engine.internal.dispatcher.controller.FieldResourcePost.handleAsync(FieldResourcePost.java:87)\n\tat io.crnk.core.engine.internal.dispatcher.controller.BaseController.handle(BaseController.java:51)","exception_class":"java.lang.NullPointerException"}`

Not sure why the `context.findAll(targetEntry, targetIds, queryContext)` in SetOwnerStrategy.java is returning null.

**Comment by remmeier** [comment]
a sanity check could be added at that place. But I guess that null is directly coming from a repository? a custom implemented one? based on line numbers it is also not a current release?

**Comment by sunmit9** [comment]
@remmeier you are right. This is not a current release (we are using 2.6.20180430102904). I just checked the file here https://github.com/crnk-project/crnk-framework/blob/c35dea80ee94dfde4e00be9c644a9f4c6e3fa144/crnk-core/src/main/java/io/crnk/core/repository/foward/strategy/SetOwnerStrategy.java. It does not seem to have handled the sanity check (in case we want to add one). 

Regarding your guess on the null coming from the custom repository, I'm not sure why the error would be random. Also, this is occurring for a first level resource which does not have any relation with other resource. 
For e.g. Drive->Folder->Files. This error occurs while create and delete Drive (It does not have any relation while creating or deleting the Drive).

**Comment by remmeier** [comment]
regarding `This error occurs while create and delete Drive (It does not have any relation while creating or deleting the Drive).` in the stacktrace above you have `FieldResourcePost`, which means that particular POSTed resource gets attached as relationship do another resource. This in turn seems to load that opposite relationship (SetOwnerStrategy) which then crashes because that repository returned null.

yes a null check should be added, it is not allowed and will crash in various places. The same thing for duplicates, such a check should be added well (just stumbled upon it lately).
